### ISSUE TITLE
Some `for` loops in typescript don't show up in context

### ISSUE THREAD

**Issue by mnzaki** [nvim-treesitter/nvim-treesitter-context#529]
### Description

`for (const a of arr)` and `for (const i in arr)` loops don't show up in context.
Only `for (let i = 0; i < n; i++)` style loops show up.

Not sure if should be reported here or in https://github.com/tree-sitter/tree-sitter-typescript
@moliva and @topaxi any ideas? since you both seem to have worked on this previously

### Neovim version

NVIM v0.10.2 Build type: RelWithDebInfo LuaJIT 2.1.1731601260

### Expected behavior

When inside a `for of` or `for in` loop that is offscreen it should show up in context

### Actual behavior

`for of` and `for in` loops don't show up in context

### Minimal config

```Lua
local plugins = {
  ts         = 'https://github.com/nvim-treesitter/nvim-treesitter',
  ts_context = 'https://github.com/nvim-treesitter/nvim-treesitter-context',
}

for name, url in pairs(plugins) do
  local install_path = '/tmp/nvim/site/'..name
  if vim.fn.isdirectory(install_path) == 0 then
    vim.fn.system { 'git', 'clone', '--depth=1', url, install_path }
  end
  vim.o.runtimepath = install_path..','..vim.o.runtimepath
end

require'nvim-treesitter.configs'.setup {
  ensure_installed = {
      "typescript",
  },
  highlight = {enable = true},
}
```

### Steps to reproduce

1. ` (echo -e 'const arr = [];\nfor (const a of arr) {'; printf '\n%.0s' {1..100}; echo '}') > test.ts`
2. `nvim --clean -u minimal.lua test.ts`
3. `<C-f>M`
4. `for of` loop is not in context!

**Comment by lewis6991** [comment]
There's not much point reporting an issue here since the queries are community maintained.

**Comment by moliva** [comment]
Essentially, you would need to add a new TreeSitter query to the language you use with the context you need. 

For reference ([1](https://github.com/nvim-treesitter/nvim-treesitter-context/blob/master/queries/typescript/context.scm#L10)) ([2](https://tree-sitter.github.io/tree-sitter/using-parsers#pattern-matching-with-queries)), you can test this in NeoVim with `:InspectTree` to find the nodes and from there `o` to toggle the query editor.

**Comment by mnzaki** [comment]
Thank you @moliva for the guidance! I managed to fix it here #530
